framework/quota/src/main/java/org/apache/cloudstack/quota/QuotaManagerImpl.java

```diff
@@ -368,7 +368,7 @@ public QuotaUsageVO updateQuotaDiskUsage(UsageVO usageRecord, final BigDecimal a
         QuotaTariffVO tariff = _quotaTariffDao.findTariffPlanByUsageType(QuotaTypes.CPU_NUMBER, usageRecord.getEndDate());
         if (tariff != null && tariff.getCurrencyValue().compareTo(BigDecimal.ZERO) != 0 && serviceoffering.getCpu() != null) {
             BigDecimal cpu = new BigDecimal(serviceoffering.getCpu());
-            onehourcostpercpu = tariff.getCurrencyValue().multiply(aggregationRatio);
```

Comment:
Hi @jairov4 The `aggregationRatio` considers the aggregation range specified in the global setting 'usage.stats.job.aggregation.range', which would be skipped when it is removed. Please check.

Reply:
@sureshanaparti, actually, considering the global setting `usage.stats.job.aggregation.range` in the calculation turns it wrong. The global setting `usage.stats.job.aggregation.range` only defines the range between each record, but the raw usage is always inserted in hours.

The `aggregationRatio` is defined by the `aggregationDuration` (`usage.stats.job.aggregation.range`) divided by `minutesInMonth` (43200). If the setting `usage.stats.job.aggregation.range` has `1440` (1 day) as value, the cost per hour will be `0.033333333`. Otherwise, if it has `60` (1 hour) as value, the cost per hour will be `0.001388889`. 

Therefore, the lower the global setting `usage.stats.job.aggregation.range` is, the lower the cost per hour will be, which is incorrect, because the raw usage is always in hour.

Reply:
@sureshanaparti considering the `aggregationRatio` here is actually causing the missbehavior. The raw usage already comes in hours as I pointed in the image above so it is independent of that ratio.